cli/cmd/install/install.go

```diff
@@ -0,0 +1,496 @@
+			return 1
+		}
+		c.kubernetes, err = kubernetes.NewForConfig(restConfig)
+		if err != nil {
+			c.UI.Output("Initializing Kubernetes client: %v", err, terminal.WithErrorStyle())
+			return 1
+		}
+	}
+
+	c.UI.Output("Pre-Install Checks", terminal.WithHeaderStyle())
+
+	err = c.checkForPreviousInstallations(settings, uiLogger)
+	if err != nil {
+		c.UI.Output(err.Error(), terminal.WithErrorStyle())
+		return 1
+	}
+
+	// Ensure there's no previous PVCs lying around.
+	err = c.checkForPreviousPVCs()
+	if err != nil {
+		c.UI.Output(err.Error(), terminal.WithErrorStyle())
+		return 1
+	}
+
+	// Ensure there's no previous bootstrap secret lying around.
+	err = c.checkForPreviousSecrets()
+	if err != nil {
+		c.UI.Output(err.Error(), terminal.WithErrorStyle())
+		return 1
+	}
+
+	// Handle preset, value files, and set values logic.
+	vals, err := c.mergeValuesFlagsWithPrecedence(settings)
+	if err != nil {
+		c.UI.Output(err.Error(), terminal.WithErrorStyle())
+		return 1
+	}
+	valuesYaml, err := yaml.Marshal(vals)
+
+	// Print out the installation summary.
+	if !c.flagAutoApprove {
+		c.UI.Output("Consul Installation Summary", terminal.WithHeaderStyle())
+		c.UI.Output("Installation name: %s", DefaultReleaseName, terminal.WithInfoStyle())
+		c.UI.Output("Namespace: %s", c.flagNamespace, terminal.WithInfoStyle())
+
+		if err != nil {
+			c.UI.Output("Overrides:"+"\n"+"%+v", err, terminal.WithInfoStyle())
+		} else if len(vals) == 0 {
+			c.UI.Output("Overrides: "+string(valuesYaml), terminal.WithInfoStyle())
+		} else {
+			c.UI.Output("Overrides:"+"\n"+string(valuesYaml), terminal.WithInfoStyle())
+		}
+	}
+
+	// Without informing the user, default global.name to consul if it hasn't been set already. We don't allow setting
+	// the release name, and since that is hardcoded to "consul", setting global.name to "consul" makes it so resources
+	// aren't double prefixed with "consul-consul-...".
+	vals = mergeMaps(convert(globalNameConsul), vals)
+
+	// Dry Run should exit here, no need to actual locate/download the charts.
+	if c.flagDryRun {
+		c.UI.Output("Dry run complete - installation can proceed.", terminal.WithInfoStyle())
+		return 0
+	}
+
+	if !c.flagAutoApprove {
+		confirmation, err := c.UI.Input(&terminal.Input{
+			Prompt: "Proceed with installation? (y/n)",
+			Style:  terminal.InfoStyle,
+			Secret: false,
+		})
+
+		if err != nil {
+			c.UI.Output(err.Error(), terminal.WithErrorStyle())
+			return 1
+		}
+		confirmation = strings.TrimSuffix(confirmation, "\n")
+		if !(strings.ToLower(confirmation) == "y" || strings.ToLower(confirmation) == "yes") {
+			c.UI.Output("Install aborted. To learn how to customize your installation, run:\nconsul-k8s install --help", terminal.WithInfoStyle())
+			return 1
+		}
+	}
+
+	c.UI.Output("Running Installation", terminal.WithHeaderStyle())
+
+	// Setup the installation action.
+	install := action.NewInstall(actionConfig)
```

Comment:
can `actionConfig` be initialized right above this step? It doesnt appear to be used elsewhere.

Reply:
Yup, moved the initialization step down!